### ISSUE TITLE
Theme Shuffling not working

### ISSUE THREAD

**Issue by BlazeUchiha420** [astronautlevel2/Anemone3DS#89]
I installed the latest version and selected the 10 themes I wanted to use the theme shuffle feature with, then installed by pressing select.Anemone says everything installed correctly. The problem is, the only theme that shows up on the home menu is the first theme I selected in the themes folder. Hitting power then home does nothing, powering down then back on also does nothing. Am I doing this wrong or is this just not working correctly?

**Comment by Helloman892** [comment]
We've had a report of this previously, actually, and I can reproduce this myself.

I'll review the code and get back to you.

**Comment by BlazeUchiha420** [comment]
Thank you very much!

**Comment by astronautlevel2** [comment]
Just tested this on the latest release (v1.2.1) and was unable to reproduce it. Can you provide additional details, and perhaps a set of test themes to try?

**Comment by ZetaDesigns** [comment]
Could also not reproduce on both latest commit and v1.2.1(o3ds)

Although it reminded me that we need to add a check if theme-management data exists, since I forgot to recreate that. Installing shuffle-themes without having that theme-management data just resets the theme.

**Comment by Helloman892** [comment]
It appears that this was actually my fault for not testing in correct conditions.

Try setting a default theme from the home menu and trying again.

**Comment by BlazeUchiha420** [comment]
It just started working for me after updating. so now it seems to be working properly

**Comment by martmists** [comment]
Can reproduce in 1.2.2 using 6 themes.

Anemone reports them to have been installed properly, but it's stuck on the one I had installed before creating a looping setup.

**Comment by Helloman892** [comment]
@martmists Can you confirm that you have switched to a default theme before trying to shuffle install? It seems to me that the theme management extdata isn't created fully simply by opening the default theme manager.

**Comment by ZetaDesigns** [comment]
To create Theme-Management extdata you need to choose the Shuffletheme option from the thememenu once.

**Comment by PlasticJustice** [comment]
I can confirm what @ZetaDesigns said. I tried installing a shuffle theme and it didn't work, but after I had selected the Shuffle Favourite Themes setting in the theme menu and shuffled a couple of the default colour themes, I was then able to install a shuffle theme thru Anemone. I believe it's because when I shuffled the default themes it updated the theme management data to allow for shuffle themes.
